### PR TITLE
wip: added / updated cluster config templates

### PR DESCRIPTION
**Pull request by masseybradley**
Adds the `ClusterRole`, `Role` and `RoleBinding` templates from #50, updated the templates names using `helm` `_helpers` and sets the default service account created to `jxl-boot` (name is generated using the fullname template).

### CODE REVIEW

**Comment by jstrachan** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rawlingsj**
You can assign the PR to them by writing `/assign @rawlingsj` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=jenkins-x-labs%2Fjxl).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/jenkins-x-labs/jxl/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by yelhouti — CHANGES_REQUESTED** [review]
@masseybradley I think the cluster-role cluster-admin is enough (and required) to allow the jx boot to install anything it wants anywhere it wants, as poeple may choose to install different apps with diffeent custoomresources in diffeent namespaces..., did you try the current version?

**Comment by jstrachan** [comment]
@masseybradley  yeah - I think we should maybe start off with `cluster-admin` to get things working then over time we can look at adding alternative roles?

**Comment by masseybradley** [comment]
@jstrachan sure, can close the PR - wouldn't mind leaving it open for a bit, will see how it goes when I'm able to clone the repo / run the install.

**Pull request closed without merging**
